diff --git a/node_modules/chessground/dist/svg.js b/node_modules/chessground/dist/svg.js
index b38c6a0..4fce1f5 100644
--- a/node_modules/chessground/dist/svg.js
+++ b/node_modules/chessground/dist/svg.js
@@ -149,6 +149,8 @@ function renderShape(state, { shape, current, hash }, brushes, dests, bounds) {
         svgs.push({ el });
         if (from[0] !== to[0] || from[1] !== to[1])
             el.appendChild(renderArrow(shape, brush, from, to, current, isShort(shape.dest, dests)));
+        else if (shape.brush === 'green')
+            el.appendChild(renderSquare(brushes.red, from, current));
         else
             el.appendChild(renderCircle(brushes[shape.brush], from, current, bounds));
     }
@@ -179,6 +181,16 @@ function renderCircle(brush, at, current, bounds) {
         r: radius - widths[1] / 2,
     });
 }
+function renderSquare(brush, at, current) {
+    return setAttributes(createElement('rect'), {
+        x: at[0] - 0.5,
+        y: at[1] - 0.5,
+        width: 1,
+        height: 1,
+        fill: brush.color,
+        'fill-opacity': 0.4,
+    });
+}
 function hilite(brush) {
     return ['#ffffff', '#fff', 'white'].includes(brush.color)
         ? hilites['hilitePrimary']
@@ -188,15 +200,15 @@ function renderArrow(s, brush, from, to, current, shorten) {
     var _a;
     function renderLine(isHilite) {
         var _a;
-        const m = arrowMargin(shorten && !current), dx = to[0] - from[0], dy = to[1] - from[1], angle = Math.atan2(dy, dx), xo = Math.cos(angle) * m, yo = Math.sin(angle) * m;
+        const m = arrowMargin(shorten && !current), originM = 6 / 64, dx = to[0] - from[0], dy = to[1] - from[1], angle = Math.atan2(dy, dx), xo = Math.cos(angle) * m, yo = Math.sin(angle) * m, xi = Math.cos(angle) * originM, yi = Math.sin(angle) * originM;
         return setAttributes(createElement('line'), {
             stroke: isHilite ? hilite(brush).color : brush.color,
             'stroke-width': lineWidth(brush, current) + (isHilite ? 0.04 : 0),
             'stroke-linecap': 'round',
             'marker-end': `url(#arrowhead-${isHilite ? hilite(brush).key : brush.key})`,
             opacity: ((_a = s.modifiers) === null || _a === void 0 ? void 0 : _a.hilite) ? 1 : opacity(brush, current),
-            x1: from[0],
-            y1: from[1],
+            x1: from[0] + xi,
+            y1: from[1] + yi,
             x2: to[0] - xo,
             y2: to[1] - yo,
         });
