diff --git a/node_modules/chessground/dist/svg.js b/node_modules/chessground/dist/svg.js
index b38c6a0..eb6f3ca 100644
--- a/node_modules/chessground/dist/svg.js
+++ b/node_modules/chessground/dist/svg.js
@@ -21,19 +21,16 @@ export function renderSvg(state, shapesEl, customsEl) {
         sources.add(moveAngle(from, to));
         dests.set(s.dest, sources);
     }
-    const shapes = d.shapes.concat(nonPieceAutoShapes).map((s) => {
+    const shapes = d.shapes
+        .concat(nonPieceAutoShapes)
+        .filter(s => s.dest)
+        .map((s) => {
         return {
             shape: s,
             current: false,
             hash: shapeHash(s, isShort(s.dest, dests), false, bounds),
         };
     });
-    if (cur)
-        shapes.push({
-            shape: cur,
-            current: true,
-            hash: shapeHash(cur, isShort(cur.dest, dests), true, bounds),
-        });
     const fullHash = shapes.map(sc => sc.hash).join(';');
     if (fullHash === state.drawable.prevSvgHash)
         return;
@@ -188,15 +185,15 @@ function renderArrow(s, brush, from, to, current, shorten) {
     var _a;
     function renderLine(isHilite) {
         var _a;
-        const m = arrowMargin(shorten && !current), dx = to[0] - from[0], dy = to[1] - from[1], angle = Math.atan2(dy, dx), xo = Math.cos(angle) * m, yo = Math.sin(angle) * m;
+        const m = arrowMargin(shorten && !current), originM = 6 / 64, dx = to[0] - from[0], dy = to[1] - from[1], angle = Math.atan2(dy, dx), xo = Math.cos(angle) * m, yo = Math.sin(angle) * m, xi = Math.cos(angle) * originM, yi = Math.sin(angle) * originM;
         return setAttributes(createElement('line'), {
             stroke: isHilite ? hilite(brush).color : brush.color,
             'stroke-width': lineWidth(brush, current) + (isHilite ? 0.04 : 0),
             'stroke-linecap': 'round',
             'marker-end': `url(#arrowhead-${isHilite ? hilite(brush).key : brush.key})`,
             opacity: ((_a = s.modifiers) === null || _a === void 0 ? void 0 : _a.hilite) ? 1 : opacity(brush, current),
-            x1: from[0],
-            y1: from[1],
+            x1: from[0] + xi,
+            y1: from[1] + yi,
             x2: to[0] - xo,
             y2: to[1] - yo,
         });
